code/osx32_syscalls.cpp

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2016-2017 Andrew Kallmeyer <fsmv@sapium.net>
+// Provided under the MIT License: https://mit-license.org
+
+#include <stddef.h>
+#include <stdint.h>
+#include <sys/syscall.h>
+#include <sys/types.h>
+
+#define IsError(err) ((uint32_t)(err) > (uint32_t)-4096)
+#define Errno(err) (-(int32_t)(err))
+
+extern "C" {
+    uint32_t syscall1(uint32_t call, void*)                                     asm("syscall1");
+    uint32_t syscall2(uint32_t call, void*, void*)                              asm("syscall2");
+    uint32_t syscall3(uint32_t call, void*, void*, void*)                       asm("syscall3");
+    uint32_t syscall4(uint32_t call, void*, void*, void*, void*)                asm("syscall4");
+    uint32_t syscall5(uint32_t call, void*, void*, void*, void*, void*)         asm("syscall5");
+    uint32_t syscall6(uint32_t call, void*, void*, void*, void*, void*, void*)  asm("syscall6");
+
+    inline void exit(int retcode) {
+        syscall1(SYS_exit, (void*)retcode);
+        __builtin_unreachable();
+    }
+
+    inline int32_t write(int fd, const void *buf, size_t length) {
+        return syscall3(SYS_write, (void*)fd, (void*)buf, (void*)length);
+    }
+
+    inline int munmap(void *addr, size_t length) {
+        return (int)syscall2(SYS_write, (void*)addr, (void*)length);
```

Comment:
Need to make sure to get this change included (in both of the _syscalls.cpp files) https://github.com/fsmv/dfre/commit/4b49b2387d768cfde0dcea47149585985f82c895

Ideally you could just rebase on top of that or something. Not sure if it works that way with git.

Reply:
Corrected with dbdf63a

Reply:
Uh oh, you made a similar mistake lol. Now you have:

munmap = write
mmap = munmap

Reply:
linking to review

Reply:
Damn, fixed in a16f2f7 😅